quasar_entities/src/event.rs

```diff
@@ -0,0 +1,119 @@
+//! `SeaORM` Entity. Generated by sea-orm-codegen 0.12.3
+use async_graphql::ComplexObject;
+use sea_orm::{entity::prelude::*, ActiveValue, Set};
+use serde_json::json;
+use stellar_xdr::{ContractEvent, Error, ScVal, WriteXdr};
+
+#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Eq, async_graphql::SimpleObject)]
+#[sea_orm(table_name = "events")]
+#[graphql(complex)]
+#[graphql(name = "Events")]
+pub struct Model {
+    #[sea_orm(primary_key)]
+    pub id: i32,
+    pub topic: String,
+    pub contract_id: String,
+    pub transaction_id: String,
+    pub value: Json,
+    pub r#type: String,
+}
+
+#[ComplexObject]
+impl Model {}
+
+#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
+pub enum Relation {}
+
+impl ActiveModelBehavior for ActiveModel {}
+
+impl TryFrom<ContractEvent> for ActiveModel {
+    type Error = Error;
+
+    fn try_from(event: ContractEvent) -> Result<Self, Self::Error> {
+        println!("{event:?}");
+        let (topic, value) = match &event.body {
+            stellar_xdr::ContractEventBody::V0(body) => {
+                let topic = &body.topics[0];
+                let topic = match topic {
+                    ScVal::Symbol(topic) => topic.to_string().unwrap(),
+                    _ => {
+                        panic!("not a symbol")
+                    }
+                };
+                let value = val_to_json(&body.data);
+                (topic, value)
+            }
+        };
+        Ok(Self {
+            id: ActiveValue::NotSet,
+            topic: ActiveValue::Set(topic),
+            contract_id: Set(stellar_strkey::Contract(event.contract_id.unwrap().0).to_string()),
+            transaction_id: ActiveValue::NotSet,
+            value: ActiveValue::Set(value),
+            r#type: Set(event.type_.to_string()),
+        })
+    }
+}
+
+fn val_to_json(val: &ScVal) -> Json {
+    match val {
```

Comment:
For readability we could merge some of the `match` arms here using `|`. For example:

```
 ScVal::U32(val) | ScVal::I32(val)| ScVal::U64(val)| ScVal::I64(val) => json!(val),
```

Reply:
These are different types, the compiler complains.
```
in the same arm, a binding must have the same type in all alternatives
```

Reply:
Ok, thanks